src/Diagnostics/windowed_spatial_average.jl

```diff
@@ -0,0 +1,47 @@
+using Statistics: mean
+using Oceananigans.Fields: compute!
+
+struct WindowedSpatialAverage{F, S, D}
+          field :: F
+   field_slicer :: S
+           dims :: D
+end
+
+"""
+    WindowedSpatialAverage(field; dims, field_slicer=FieldSlicer())
+
+Builds a `WindowedSpatialAverage` of `field` that returns the average of `field` over a 
+
+slice defined by `field_sicer`.
+
+
+Example
+=======
+
+```julia
+using Oceananigans.Diagnostics: WindowedSpatialAverage
+
+slicer = FieldSlicer(j=Ny÷2+1:Ny)
+
+U_wsa = WindowedSpatialAverage(u; dims=(1, 2), field_slicer=slicer)
+
+simulation.output_writers[:simple_output] = NetCDFOutputWriter(model, (U_wsa=U_wsa,), 
+                                                               schedule = AveragedTimeInterval(10seconds),
+                                                               filepath = "file.nc")
+```
```

Comment:
Might be good to convert this to `jldoctest` block so it's tested when docs are built (to avoid the example going out of date)?

Reply:
Sounds good! I didn't know we could do that. Is it enough to substitute `julia` for `jldoctest`? (And then of course create a model, simulation, etc?)

Reply:
Okay, took a crack at it. Although I'm not sure how the docs deal with the creation of the output file `"file.nc"`. Should I remove that line?

Reply:
The new file is innocuous because its created somewhere when the docs are built and then destroyed. It does clutter up your `docs/` if you build locally. I don't think this is the only culprit. There's things like `stuff.nc` lying around as well. But we could use a better name for the file like `windowed_spatial_average_example.nc` so its easier to track down in the future.

Reply:
Ah we might be able to use a `@meta` block after the `jldoctest` to delete the file: https://juliadocs.github.io/Documenter.jl/v0.7/man/doctests.html#Setup-Code-1

So something like like this might work?

````
```@meta
DocTestSetup = quote
    rm("windowed_spatial_average_example.nc")
end
```
````

Reply:
EDIT: I suggested a `@meta` block with `DocTestSetup` might let us delete the file right after the doctest but after reading the Documenter.jl docs a bit more, it turns out it only gets executed before a `jldoctest`, not after.

Reply:
> But we could use a better name for the file like windowed_spatial_average_example.nc so its easier to track down in the future.

Good point. Done!